downstairs/src/lib.rs

```diff
@@ -1241,15 +1241,25 @@ where
              */
             _ = sleep_until(lossy_interval), if lossy => {
                 job_channel_tx.send(()).await?;
-                lossy_interval = deadline_secs(5);
+                lossy_interval = deadline_secs(5.0);
             }
             /*
-             * Don't wait more than 50 seconds to hear from the other side.
+             * Don't wait more than TIMEOUT_SECS * TIMEOUT_LIMIT seconds to hear
+             * from the other side.
              * XXX Timeouts, timeouts: always wrong!  Some too short and
              * some too long.
              */
-            _ = sleep_until(deadline_secs(50)) => {
-                bail!("inactivity timeout");
+            _ = sleep_until(deadline_secs(TIMEOUT_SECS)) => {
```

Comment:
I'm wondering if this is the same behaviour as what we have in the Upstairs: we have a variable `timeout = deadline_secs(TIMEOUT_SECS);`, which is bumped when we receive a new message, where her it's on the tokio select expression, recomputed each loop.

Reply:
I had the same thought, but believe the behavior is identical in both places: in the Upstairs, all branches either reset the timeout or break out of the loop.  That's equivalent to starting the timeout from 0 on each iteration through the loop.

(also, there's no reason to use `sleep_until(deadline_secs(..))` versus plain `sleep(..)`)

Reply:
However, thinking about it again, the **downstairs** behavior may be wrong!  Right now, if `lossy` is set, then we'll never timeout, because `lossy_interval < TIMEOUT_SECS`

Reply:
Oh, if hitting the lossy timeout reset the whole select, then that issue has been.. here.. for a while now.
I guess because we never really test both lossy and timeouts, we never noticed.

Reply:
Turn out the lossy in this select was not even needed.  The way the `do_work_for()` loop is structured,
it will always run to completion (skipped jobs are put on the end).  So, it's no longer needed to have
lossy make the call it was doing here.

I've removed it in a76fbc1